server/src/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java

```diff
@@ -1046,4 +1046,51 @@ private VmWorkJobVO createPlaceHolderWork(long instanceId) {
 
         return workJob;
     }
+
+    @Override
+    public void cleanupVmSnapshotJobs(){
+        List<AsyncJobVO> jobs = _jobMgr.findFailureAsyncJobs(VmWorkCreateVMSnapshot.class.getName(), VmWorkRevertToVMSnapshot.class.getName());
+
+        for (AsyncJobVO job : jobs) {
+            try {
+                if (job.getCmd().equalsIgnoreCase(VmWorkCreateVMSnapshot.class.getName())) {
+                    VmWorkCreateVMSnapshot work = VmWorkSerializer.deserialize(VmWorkCreateVMSnapshot.class, job.getCmdInfo());
+                    cleanupVmSnapshotCreateFailure(work.getVmSnapshotId());
+                } else if(job.getCmd().equalsIgnoreCase(VmWorkRevertToVMSnapshot.class.getName())) {
+                    VmWorkRevertToVMSnapshot work = VmWorkSerializer.deserialize(VmWorkRevertToVMSnapshot.class, job.getCmdInfo());
+                    cleanupVmSnapshotRevertFailure(work.getVmSnapshotId());
+                }
+            } catch (Exception e) {
```

Comment:
what exception do you expect here? Is it the deserialization failure or any thing else? Can you catch specific exceptions?